### PR TITLE
🐛Support fullscreening <video>s in <amp-twitter>

### LINKED ISSUE #15650: amp-twitter: embedded videos can not go fullscreen
Sample to repro:
```
<amp-twitter width="375"
  height="472"
  layout="responsive"
  data-tweetid="885634330868850689">
</amp-twitter>
```

This tweet has a video but `fullscreen` video UI control is not available for it. 

Likely due to the `iframe` not allowing it.

### PR DESCRIPTION
**Pull request by sparhami**
- Replaces `getIframe`'s opt_disallowCustom with an options object argument
- Make an explicit API for `allowfullscreen` to encourage users of `getIframe` to consider whether they need fullscreen support
- Adds the `fullscreen` attribute to the iframe created for `<amp-twitter>`

Fixes #15650

### CODE REVIEW

File: src/3p-frame.js [hunk 1]
```diff
@@ -60,18 +60,36 @@ function getFrameAttributes(parentWindow, element, opt_type, opt_context) {
   return attributes;
 }
 
+/** @record */
+class IframeOptions {
```
Comment (aghassemi) [inline src/3p-frame.js]: we usually just define the type in place ` @param {{disallowCustom:..}} options` or use typedef and name them `/** @typedef {{disallowCustom:..}} */ let IframeOptionsDef;`

File: src/3p-frame.js [hunk 2]
```diff
@@ -60,18 +60,36 @@ function getFrameAttributes(parentWindow, element, opt_type, opt_context) {
...
+
 /**
  * Creates the iframe for the embed. Applies correct size and passes the embed
  * attributes to the frame via JSON inside the fragment.
  * @param {!Window} parentWindow
  * @param {!AmpElement} parentElement
  * @param {string=} opt_type
  * @param {Object=} opt_context
- * @param {boolean=} opt_disallowCustom whether 3p url should not use meta tag.
+ * @param {!IframeOptions=} options Options for the created iframe.
```
Comment (aghassemi) [inline src/3p-frame.js]: `opt-options`? sounds redundant but consistent.

File: extensions/amp-twitter/0.1/amp-twitter.js [hunk 3]
```diff
@@ -59,7 +59,8 @@ class AmpTwitter extends AMP.BaseElement {
 
   /** @override */
   layoutCallback() {
-    const iframe = getIframe(this.win, this.element, 'twitter');
+    const iframe = getIframe(this.win, this.element, 'twitter', null,
```
Comment (aghassemi) [inline extensions/amp-twitter/0.1/amp-twitter.js]: let's do `amp-reddit` too, just realized it has the same bug.

File: src/3p-frame.js (hunk 1 above)
Comment (sparhami) [reply src/3p-frame.js]: Changed to just inline it. The one limitation is that you can't comment the parameters :(. Maybe better parameter names would help?

File: extensions/amp-twitter/0.1/amp-twitter.js (hunk 3 above)
Comment (sparhami) [reply extensions/amp-twitter/0.1/amp-twitter.js]: Done.

File: src/3p-frame.js (hunk 2 above)
Comment (sparhami) [reply src/3p-frame.js]: Changed for consistency. The [google style guide](https://google.github.io/styleguide/jsguide.html#features-functions-default-parameters) mentions they should not be present, but will keep the file consistent. QQ: For new files, should they try to match the style guide or keep with AMP conventions?

**Comment by aghassemi** [comment]
Looks good. There is a lint failure (we recently started running new lint rules on any files change, so error is not related to your code but since you touched a file, it is now needed to be fixed)

File: src/3p-frame.js (hunk 2 above)
Comment (aghassemi) [reply src/3p-frame.js]: AMP convention is sometimes-badly-followed Google style guide :) so for new files, stick to the guid

**Comment by aghassemi** [comment]
Help page: https://support.google.com/analytics/answer/7486764?hl=en

**Comment by aghassemi** [comment]
@sparhami merged 🎉

**Pull request merged**